### ISSUE TITLE
npm install fails with "Cannot read property 'target' of null"

### ISSUE THREAD

**Issue by ackalker** [npm/npm#11941]
```sh
$ npm version
{ npm: '3.8.1',
  ares: '1.10.1-DEV',
  http_parser: '2.6.2',
  icu: '56.1',
  modules: '47',
  node: '5.9.0',
  openssl: '1.0.2g',
  uv: '1.8.0',
  v8: '4.6.85.31',
  zlib: '1.2.8' }
```

The idea is to install a [Tern](https://www.npmjs.com/package/tern) plugin, [`tern-node-extension`](https://www.npmjs.com/package/tern-node-extension), for which the documentation instructs to install it in the directory in which Tern is installed.

Because I want to setup my NodeJS development environment programmatically, for easy deployment on multiple machines, I use a script whose actions amount to the following actions, minimized to installing the mentioned package and plugin. It can serve as a reproducible test case:

```sh
rm ~/.npmrc
npm set prefix ~/.local
npm set python python2
npm set init.author.name "<hidden>"
npm set init.author.email "<hidden>"
...
npm install -g tern
npm explore -g tern -- npm install tern-node-extension
```

The same error message appears when I manually remove and redo npm's config, then do:
```
...
$ npm install -g tern
/home/miki/.local/bin/tern -> /home/miki/.local/lib/node_modules/tern/bin/tern
/home/miki/.local/lib
`-- tern@0.18.0 
  +-- acorn@2.7.0 
  +-- enhanced-resolve@0.9.1 
  | +-- graceful-fs@4.1.3 
  | +-- memory-fs@0.2.0 
  | `-- tapable@0.1.10 
  +-- glob@3.2.11 
  | +-- inherits@2.0.1 
  | `-- minimatch@0.3.0 
  +-- minimatch@0.2.14 
  | +-- lru-cache@2.7.3 
  | `-- sigmund@1.0.1 
  `-- typescript@1.0.1 

$ npm explore -g tern

Exploring /home/miki/.local/lib/node_modules/tern
Type 'exit' or ^D when finished

$ npm install tern-node-extension
npm ERR! Linux 4.4.5-1-ARCH
npm ERR! argv "/usr/bin/node" "/usr/bin/npm" "install" "tern-node-extension"
npm ERR! node v5.9.0
npm ERR! npm  v3.8.1

npm ERR! Cannot read property 'target' of null
npm ERR! 
npm ERR! If you need help, you may report this error at:
npm ERR!     <https://github.com/npm/npm/issues>

npm ERR! Please include the following file with any support request:
npm ERR!     /home/miki/.local/lib/node_modules/tern/npm-debug.log
$ exit
```

[Here](https://gist.github.com/c212c998e8f611d61dd7) is the `npm-debug.log` produced from the command above.

**Comment by ackalker** [comment]
Trying to install `tern-node-extension` after manually changing to the directory in which `tern` is installed produces the same error:
```sh
$ cd ~/.local/lib/node_modules/tern/
$ npm install tern-node-extension
npm ERR! Linux 4.4.5-1-ARCH
npm ERR! argv "/usr/bin/node" "/usr/bin/npm" "install" "tern-node-extension"
npm ERR! node v5.9.0
npm ERR! npm  v3.8.1

npm ERR! Cannot read property 'target' of null
npm ERR! 
npm ERR! If you need help, you may report this error at:
npm ERR!     <https://github.com/npm/npm/issues>

npm ERR! Please include the following file with any support request:
npm ERR!     /home/miki/.local/lib/node_modules/tern/npm-debug.log
```

**Comment by MoOx** [comment]
Seems like a duplicate of #10686.

**Comment by othiym23** [comment]
Indeed. Closing as duplicate of #10686, which may or may not be fixed by #11947, so give `npm@next` a look if you run into this. Thanks for your time and patience!

**Comment by ackalker** [comment]
@othiym23 I'm sorry to say, using `npm` installed from commit 473d3244a8ddfd6b260d0aa0d395b119d595bf97 gives a [different error](https://gist.github.com/a6471b06f6d36073d20f) (same with HEAD @ f2ae3ba).
Please see my bisection result in https://github.com/npm/npm/issues/10686#issuecomment-200353028 , the problem may be elsewhere.
